### ISSUE TITLE
is_nvidia_device() broken on Mac OS X 10.9.5

### ISSUE THREAD

**Issue by msuchard** [boostorg/compute#471]
The following code:

```{c++}
#include <iostream>

#include <boost/compute/core.hpp>
#include <boost/compute/detail/vendor.hpp>

namespace compute = boost::compute;

int main()
{
    // get the default device
    compute::device device = compute::system::default_device();

    // print the device's vendor
    std::cout << device.vendor() << std::endl;

    std::cout << compute::detail::is_nvidia_device(device) << std::endl;

    return 0;
}
```

produces
```
NVIDIA
0
```

since
```{c++}
return device.vendor() == "NVIDIA Corporation"
````

fails in `vendor.hpp`.  Am happy to provide a pull-request fix.

best, Marc

**Comment by pavanky** [comment]
Perhaps the default device on your machine is the intel GPU?

**Comment by kylelutz** [comment]
Fixed in PR #472.
